### PR TITLE
Basic support for maintenance mode of mesos

### PR DESCRIPTION
**Pull request by zmalik**
As mentioned in #163 the basic support to reject offers if the node is currently in maintenance window.

### CODE REVIEW

**Comment by alde** [comment]
looks good to me, thanks for the PR

**Comment by zmalik** [comment]
Discovered a bug while testing with mesos, so now a small fix as I was interpreting wrong the DurationInfo in last commit.

File: mesos/match.go [hunk 1]
```diff
@@ -68,6 +77,28 @@ func (m *attributeMatcher) Matches(o interface{}) error {
...
 }
 
+func (m *availabilityMatcher) Matches(o interface{}) error {
+	offer := o.(*mesosproto.Offer)
+
+	if offer.Unavailability == nil {
+		return nil
+	}
+
+	now := time.Now().UnixNano()
```
Comment (keis) [inline mesos/match.go]: A bit of a nitpick but timestamps always puts my in worry about everything mode. So I think getting the current timestamp is better to do when we create the matcher (sent as an argument even) so the implementation can remain completely pure.

File: mesos/match_test.go [hunk 2]
```diff
@@ -58,6 +62,54 @@ func TestMatch(t *testing.T) {
...
+		},
+		&mesosproto.Attribute{
+			Name: proto.String("node_name"),
+			Type: mesosproto.Value_TEXT.Enum(),
+			Text: &mesosproto.Value_Text{
+				Value: proto.String("node4"),
+			},
+		},
+	)
+	offerE := offer("offer-c", 1.8, 512.0,
```
Comment (keis) [inline mesos/match_test.go]: should be "offer-e" in the id-string

Comment (zmalik) [inline mesos/match_test.go]: thanks!

File: mesos/match.go (hunk 1 above)
Comment (zmalik) [inline mesos/match.go]: 👌

**Comment by keis** [comment]
Thanks a lot @zmalik

**Pull request merged**